File: schemas/sdg_coin_wallet.py
from datetime import datetime
from typing import List, Union

from pydantic import BaseModel


class SDGCoinWalletSchemaBase(BaseModel):
    sdg_coin_wallet_id: int
    user_id: int
    total_coins: float
    histories: List[Union["SDGCoinWalletHistorySchemaBase", "SDGCoinWalletHistorySchemaFull"]]

    model_config = {
        "from_attributes": True
    }


class SDGCoinWalletSchemaFull(SDGCoinWalletSchemaBase):
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend/types/sdgCoinWallet.ts
import {
  SDGCoinWalletHistorySchemaBase,
  SDGCoinWalletHistorySchemaFull,
} from "./sdgCoinWalletHistory";

export interface SDGCoinWalletSchemaBase {
  sdgCoinWalletId: number;
  userId: number;
  totalCoins: number;
  histories: (SDGCoinWalletHistorySchemaBase | SDGCoinWalletHistorySchemaFull)[];
}

export interface SDGCoinWalletSchemaFull extends SDGCoinWalletSchemaBase {
  createdAt: string;
  updatedAt: string;
}
